x-pack/test/security_api_integration/tests/kerberos/kerberos_login.ts

```diff
@@ -142,26 +143,29 @@ export default function ({ getService }: FtrProviderContext) {
           ? ['kibana_admin', 'superuser_anonymous']
           : ['kibana_admin'];
 
-        await supertest
+        const spnegoResponse = await supertest
           .get('/internal/security/me')
           .set('kbn-xsrf', 'xxx')
           .set('Cookie', sessionCookie.cookieString())
-          .expect(200, {
-            username: 'tester@TEST.ELASTIC.CO',
-            roles: expectedUserRoles,
-            full_name: null,
-            email: null,
-            metadata: {
-              kerberos_user_principal_name: 'tester@TEST.ELASTIC.CO',
-              kerberos_realm: 'TEST.ELASTIC.CO',
-            },
-            enabled: true,
-            authentication_realm: { name: 'kerb1', type: 'kerberos' },
-            lookup_realm: { name: 'kerb1', type: 'kerberos' },
-            authentication_provider: { type: 'kerberos', name: 'kerberos' },
-            authentication_type: 'token',
-            elastic_cloud_user: false,
-          });
+          .expect(200);
+
+        jestExpect(spnegoResponse.body).toEqual({
+          username: 'tester@TEST.ELASTIC.CO',
+          roles: expectedUserRoles,
+          full_name: null,
+          email: null,
+          metadata: {
+            kerberos_user_principal_name: 'tester@TEST.ELASTIC.CO',
+            kerberos_realm: 'TEST.ELASTIC.CO',
+          },
+          enabled: true,
+          authentication_realm: { name: 'kerb1', type: 'kerberos' },
+          lookup_realm: { name: 'kerb1', type: 'kerberos' },
+          authentication_provider: { type: 'kerberos', name: 'kerberos' },
+          authentication_type: 'token',
+          elastic_cloud_user: false,
+          profile_uid: jestExpect.any(String),
```

Comment:
__note:__ Since the user profile UID isn't part of the ES authentication response and we have a custom logic to enrich user with it, it'd be great if instead of `jestExpect.any(String)` here and in similar API integration tests for other major authentication providers, we could check that profile uid is indeed correct (e.g. fetching it from `/internal/security/user_profile` before this assertion).

What do you think?

Reply:
I agree that it's worth having an end to end test for the enrichment logic but I don't think it's correct to clutter this in with the authentication providers since the logic is independent of provider logic. I have added an end to end test for the enrichment logic to the user profile related functional test suite instead.